src/main/java/seedu/phu/ui/CommandBox.java

```diff
@@ -42,13 +49,54 @@ private void handleCommandEntered() {
         }
 
         try {
+            inputHistory.addInput(commandText);
             commandExecutor.execute(commandText);
             commandTextField.setText("");
         } catch (CommandException | ParseException e) {
             setStyleToIndicateCommandFailure();
         }
     }
 
+    /**
+     * Handles UP and DOWN key pressed event.
+     */
+    @FXML
+    private void handleKeyClick(KeyEvent keyEvent) {
+        KeyCode clickedKey = keyEvent.getCode();
+
+        if (clickedKey.equals(KeyCode.UP)) {
+            keyEvent.consume();
+            fetchPreviousInput();
+        } else if (clickedKey.equals(KeyCode.DOWN)) {
+            keyEvent.consume();
+            fetchNextInput();
+        }
+    }
```

Comment:
Currently, the caret will move to the start of the text input when clicking the up / down key (except on reaching the last command). I think it will be more convenient for the users if you set it to the end of the current input and to make it consistent.

I think you can use the [`positionCaret`](https://docs.oracle.com/javase/8/javafx/api/javafx/scene/control/TextInputControl.html#positionCaret-int- ) method to do this.